dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java

```diff
@@ -268,25 +276,42 @@ private Tuple2<String, List<Object>> selectQuery(final BrowserQuery browserQuery
         final StringBuilder sqlQuery = new StringBuilder("select cvi." + workingLiveInode + " as inode "
                 + " from contentlet_version_info cvi, identifier id, structure struc, contentlet c "
                 + " where cvi.identifier = id.id and struc.velocity_var_name = id.asset_subtype and  "
-                + " c.inode = cvi." + workingLiveInode);
+                + " c.inode = cvi." + workingLiveInode + " and cvi.variant_id='"+DEFAULT_VARIANT.name()+"' ");
+
+
+        final StringBuilder luceneQuery = UtilMethods.isSet(browserQuery.luceneQuery) ? new  StringBuilder("+title:*" + browserQuery.luceneQuery.trim() + "* ") : new  StringBuilder();
+
+        if(browserQuery.showWorking || browserQuery.showArchived) {
+            luceneQuery.append("+working:true ");
+        }
+        else {
+            luceneQuery.append("+live:true ");
+        }
+        if(browserQuery.showArchived ) luceneQuery.append("+deleted:true ");
 
         if (!showAllBaseTypes) {
             List<String> baseTypes =
                     browserQuery.baseTypes.stream().map(t -> String.valueOf(t.getType())).collect(Collectors.toList());
             sqlQuery.append(" and struc.structuretype in (" + String.join(" , ", baseTypes) + ") ");
+
+            luceneQuery.append("+baseType:(" + String.join(" OR ", baseTypes)+ ") ");
         }
         if (browserQuery.languageId > 0) {
             sqlQuery.append(" and cvi.lang = ? ");
             parameters.add(browserQuery.languageId);
+            luceneQuery.append("+languageId:" + browserQuery.languageId + " ");
         }
         if (browserQuery.site != null) {
-            sqlQuery.append(" and id.host_inode = ? ");
+            sqlQuery.append(" and (id.host_inode = ? OR id.host_inode = '"+ Host.SYSTEM_HOST +"') ");
             parameters.add(browserQuery.site.getIdentifier());
+            luceneQuery.append("+conHost:("+ Host.SYSTEM_HOST +" OR " + browserQuery.site.getIdentifier() + ") ");
```

Comment:
It seems like you're not getting the most out of using a StringBuilder because appending many strings with `+` instead of using `.append` makes string management expensive

Reply:
you are right! @jcastro-dotcms made some improvements in the code